spring-web/src/main/java/org/springframework/http/client/HttpComponentsClientHttpRequestFactory.java

```diff
@@ -90,8 +90,7 @@ public HttpComponentsClientHttpRequestFactory(HttpClient httpClient) {
 	 * Set the {@code HttpClient} used for
 	 * {@linkplain #createRequest(URI, HttpMethod) synchronous execution}.
 	 */
-	public void setHttpClient(HttpClient httpClient) {
-		Assert.notNull(httpClient, "HttpClient must not be null");
+	public void setHttpClient(@Nullable HttpClient httpClient) {
```

Comment:
The field should change to not-null along with direct field population in the constructors here. I'll commit this in a bit.